core/arch/arm/mm/core_mmu.c

```diff
@@ -438,13 +431,74 @@ uint32_t core_mmu_type_to_attr(enum teecore_memtypes t)
 	}
 }
 
+static bool __maybe_unused map_is_tee_ram(struct tee_mmap_region *mm)
+{
+	return mm->type == MEM_AREA_TEE_RAM;
+}
+
+static bool map_is_flat_mapped(struct tee_mmap_region *mm)
+{
+	return map_is_tee_ram(mm);
+}
+
+static bool __maybe_unused map_is_secure(struct tee_mmap_region *mm)
+{
+	return !!(core_mmu_type_to_attr(mm->type) & TEE_MATTR_SECURE);
+}
+
+static void swap_mmap(struct tee_mmap_region *mm1, struct tee_mmap_region *mm2)
+{
+	struct tee_mmap_region swap;
+
+	if (mm1 == mm2)
+		return;
+
+	swap = *mm1;
+	*mm1 = *mm2;
+	*mm2 = swap;
+}
+
+static void dump_mmap_table(struct tee_mmap_region *memory_map)
+{
+	struct tee_mmap_region *map;
+
+	for (map = memory_map; map->type != MEM_AREA_NOTYPE; map++) {
+		vaddr_t __maybe_unused vstart;
+
+		vstart = map->va + ((vaddr_t)map->pa & (map->region_size - 1));
+		DMSG("type va %d 0x%08" PRIxVA "..0x%08" PRIxVA
+		     " pa 0x%08" PRIxPA "..0x%08" PRIxPA " size %#zx",
+		     map->type, vstart, vstart + map->size - 1,
+		     map->pa, (paddr_t)(map->pa + map->size - 1),
+		     map->size);
+	}
+}
+
+static void sort_mmap_table_by_va(struct tee_mmap_region *memory_map)
```

Comment:
An alternative could be to just use `qsort()` instead, until there's a problem with it.